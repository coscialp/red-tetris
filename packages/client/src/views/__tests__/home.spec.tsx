import { render, screen } from "@testing-library/react";
import Home from "../home/home.tsx";
import { Provider } from "react-redux";
import { store } from "../../stores";


// Mock useDispatch
jest.mock('react-redux', () => ({
  ...(jest.requireActual('react-redux') as object),
  useDispatch: jest.fn(),
}));

describe("Home", () => {
  beforeEach(() => {
    // Reset mock implementation before each test
    jest.clearAllMocks();
  });

  it("renders Home component", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    // Add assertions for the content rendered by Home component
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays start button when appropriate", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    const button = screen.queryByTestId("start-button");
    expect(button).toBeDefined();
  });
});