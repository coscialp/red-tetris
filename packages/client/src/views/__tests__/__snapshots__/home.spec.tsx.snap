// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: center; align-items: center; height: 100vh;"
  >
    <div
      class="Toastify"
    >
      <div
        class="Toastify__toast-container Toastify__toast-container--top-right"
      >
        <div
          class="Toastify__toast Toastify__toast-theme--dark Toastify__toast--error Toastify__toast--close-on-click Toastify--animate Toastify__bounce-enter--top-right"
          data-in="true"
          id="2"
        >
          <div
            class="Toastify__toast-body"
            role="alert"
          >
            <div
              class="Toastify__toast-icon Toastify--animate-icon Toastify__zoom-enter"
            >
              <svg
                fill="var(--toastify-icon-color-error)"
                height="100%"
                viewBox="0 0 24 24"
                width="100%"
              >
                <path
                  d="M11.983 0a12.206 12.206 0 00-8.51 3.653A11.8 11.8 0 000 12.207 11.779 11.779 0 0011.8 24h.214A12.111 12.111 0 0024 11.791 11.766 11.766 0 0011.983 0zM10.5 16.542a1.476 1.476 0 011.449-1.53h.027a1.527 1.527 0 011.523 1.47 1.475 1.475 0 01-1.449 1.53h-.027a1.529 1.529 0 01-1.523-1.47zM11 12.5v-6a1 1 0 012 0v6a1 1 0 11-2 0z"
                />
              </svg>
            </div>
            <div>
              Invalid URL
            </div>
          </div>
          <button
            aria-label="close"
            class="Toastify__close-button Toastify__close-button--dark"
            type="button"
          >
            <svg
              aria-hidden="true"
              viewBox="0 0 14 16"
            >
              <path
                d="M7.71 8.23l3.75 3.75-1.48 1.48-3.75-3.75-3.75 3.75L1 11.98l3.75-3.75L1 4.48 2.48 3l3.75 3.75L9.98 3l1.48 1.48-3.75 3.75z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="Toastify__progress-bar--wrp"
            data-hidden="false"
          >
            <div
              class="Toastify__progress-bar--bg Toastify__progress-bar-theme--dark Toastify__progress-bar--error"
            />
            <div
              aria-hidden="false"
              aria-label="notification timer"
              class="Toastify__progress-bar Toastify__progress-bar--animated Toastify__progress-bar-theme--dark Toastify__progress-bar--error"
              role="progressbar"
              style="animation-duration: 5000ms; animation-play-state: paused;"
            />
          </div>
        </div>
      </div>
    </div>
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
